### DISCUSSION TITLE
Entanglement Entropy of Ising Model H=-JZZ-hX

### DISCUSSION THREAD

**Discussion by hcut** [QuSpin/QuSpin#711] [Q&A]
Entanglement entropy of  Ising model  $H=-JZZ-hX$ 

![3](https://github.com/user-attachments/assets/b1aa39fb-1ff0-4af7-8598-b72f84c651e4)

is given as (for $J=h=1, c=0.5$) for subsystem x, S(x) =

![1](https://github.com/user-attachments/assets/093e9328-7096-486d-8efe-6b808b0009c0)

Which looks  like as 

![2](https://github.com/user-attachments/assets/f792bc82-c3c1-4234-8b3a-78414f456efc)

I am trying to reproduce the same results, but I am not getting even close to this figures, I am getting the following figure

![4](https://github.com/user-attachments/assets/07613792-208a-4676-b7b1-3ce90ca44605)

I am calculating as follows : 

     ...
     e_m,v_m=H.eigsh(k=1,which="SA",maxiter=1E4,pars=dict(h0=1.0,h1=H_x),return_eigenvectors=True)
     def get_ee(j):
            subsys=[i for i in range(j)] 
            Sent_dict = ent_entropy(v_m, basis, chain_subsys=subsys, return_rdm='A')  
            ee = Sent_dict['Sent_A']
           return ee 
    #
    for n in range(L):   
            ee=get_ee(n)       
         #   print(ee)

 It prints the subsytems as

         [0]
         [0, 1]
         [0, 1, 2]
         [0, 1, 2, 3]
         [0, 1, 2, 3, 4]
         ........
         

I think I am doing something wrong in defining the subsystems at each step. 

I will be grateful in somebody can help me here ?

**Comment by jaredjeya** [comment]
It looks like you're defining the subsystem fine. However, by default the entanglement entropy is computed _per site_, not in total, so you need to pass in `density=False`. That might fix it.

However, I can't see how you've defined your `H` or `basis` so I have no idea if there's a bug there as well.

**Comment by hcut** [comment]
@jaredjeya, thanks, here it is: 


                    basis = spin_basis_1d(L,S='1/2',pauli=False)
                    s_zz_term = [[J_zz,i,(i+1)%L] for i in range(L)]
                    opr_term_0=[["zz",s_zz_term]] 
                    h_x_term=[[-1.0,i] for i in range(L)] #*J_1 
                    opr_term_1=[["x",h_x_term]]
                    op_dict=dict(h0=opr_term_0,h1=opr_term_1)
                    H=quantum_operator(op_dict,basis=basis,dtype=np.float32,check_symm=False)

It is fine ?

I am getting the following figure now  for `N=10` after passing `density=False`

![ee(10,1 0)](https://github.com/user-attachments/assets/61913185-196d-411b-8bd8-5456d0156641)




Can you suggest some document that will help in doing/understanding  the entanglement calculations ( e.g. calculating site entropy, bond entropy, total entropy ...) with Quspin ?


I will be thankful for you help.
